File: src/main/java/it/frob/sleighidea/psi/SleighTokenType.java
// SPDX-License-Identifier: Apache-2.0

package it.frob.sleighidea.psi;

import com.intellij.psi.tree.IElementType;
import it.frob.sleighidea.SleighLanguage;
import org.jetbrains.annotations.NonNls;
import org.jetbrains.annotations.NotNull;

public class SleighTokenType extends IElementType {
    public SleighTokenType(@NotNull @NonNls String debugName) {
        super(debugName, SleighLanguage.INSTANCE);
    }

    @Override
    public String toString() {
        return "SleighTokenType." + super.toString();
    }
}

File: src/main/kotlin/it/frob/sleighidea/psi/SleighTokenType.kt
// SPDX-License-Identifier: Apache-2.0

package it.frob.sleighidea.psi

import com.intellij.psi.tree.IElementType
import it.frob.sleighidea.SleighLanguage
import org.jetbrains.annotations.NonNls

class SleighTokenType(debugName: @NonNls String) : IElementType(debugName, SleighLanguage.INSTANCE) {
    override fun toString(): String = "SleighTokenType." + super.toString()
}